pkg/controllers/machinemigration/machine_migration_controller_test.go

```diff
@@ -0,0 +1,693 @@
+							Build()
+						mapiMachine.Status.AuthoritativeAPI = updatedMAPIMachine.Status.AuthoritativeAPI
+						mapiMachine.Status.Conditions = updatedMAPIMachine.Status.Conditions
+					})).Should(Succeed())
+
+					req = reconcile.Request{NamespacedName: client.ObjectKeyFromObject(mapiMachine)}
+				})
+
+				It("should request pausing for the old authoritative resource (MAPI) and stay in Migrating status", func() {
+					_, err := reconciler.Reconcile(ctx, req)
+					Expect(err).NotTo(HaveOccurred())
+
+					updatedM := &machinev1beta1.Machine{}
+					Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(mapiMachine), updatedM)).To(Succeed())
+					// To check for requesting pausing on the the MAPI resource it is sufficient
+					// see that the spec.AuthoritativeAPI field is set to ClusterAPI,
+					// which is already done anyway on the requester side.
+					Expect(updatedM.Spec.AuthoritativeAPI).To(Equal(machinev1beta1.MachineAuthorityClusterAPI))
+					Expect(updatedM.Status.AuthoritativeAPI).To(Equal(machinev1beta1.MachineAuthorityMigrating))
+				})
+			})
+			Context("when migrating from ClusterAPI to MachineAPI", func() {
+				BeforeEach(func() {
+					By("Setting the MAPI machine spec AuthoritativeAPI to MachineAPI")
+					mapiMachine = mapiMachineBuilder.WithAuthoritativeAPI(machinev1beta1.MachineAuthorityMachineAPI).Build()
+					Eventually(k8sClient.Create(ctx, mapiMachine)).Should(Succeed())
+
+					By("Creating a mirror CAPI machine")
+					capiMachine = capiMachineBuilder.Build()
+					Eventually(k8sClient.Create(ctx, capiMachine)).Should(Succeed())
+					capaMachine = capaMachineBuilder.Build()
+					Eventually(k8sClient.Create(ctx, capaMachine)).Should(Succeed())
+
+					By("Setting the MAPI machine status AuthoritativeAPI to 'Migrating'")
+					Eventually(k.UpdateStatus(mapiMachine, func() {
+						updatedMAPIMachine := mapiMachineBuilder.
+							WithAuthoritativeAPIStatus(machinev1beta1.MachineAuthorityMigrating).
+							WithConditions([]machinev1beta1.Condition{{
+								Type:               consts.SynchronizedCondition,
+								LastTransitionTime: metav1.Now(),
+								Status:             corev1.ConditionTrue}}).
+							Build()
+						mapiMachine.Status.AuthoritativeAPI = updatedMAPIMachine.Status.AuthoritativeAPI
+						mapiMachine.Status.Conditions = updatedMAPIMachine.Status.Conditions
+					})).Should(Succeed())
+
+					req = reconcile.Request{NamespacedName: client.ObjectKeyFromObject(mapiMachine)}
+				})
+
+				It("should request pausing for the old authoritative resource (CAPI) and stay in Migrating status", func() {
+					_, err := reconciler.Reconcile(ctx, req)
+					Expect(err).NotTo(HaveOccurred())
+
+					updatedM := &machinev1beta1.Machine{}
+					Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(mapiMachine), updatedM)).To(Succeed())
+					Expect(updatedM.Status.AuthoritativeAPI).To(Equal(machinev1beta1.MachineAuthorityMigrating))
+
+					updatedCAPIMS := &capiv1beta1.Machine{}
```

Comment:
MS?

Should we check the infra machine too?

Reply:
Yes we should, added.